src/Core/src/Handlers/Page/PageHandler.iOS.cs

```diff
@@ -5,16 +5,15 @@ namespace Microsoft.Maui.Handlers
 {
 	public partial class PageHandler : ContentViewHandler, INativeViewHandler
 	{
-		PageViewController? _pageViewController;
 		protected override ContentView CreateNativeView()
 		{
 			_ = VirtualView ?? throw new InvalidOperationException($"{nameof(VirtualView)} must be set to create a LayoutView");
 			_ = MauiContext ?? throw new InvalidOperationException($"{nameof(MauiContext)} cannot be null");
 
-			_pageViewController ??= new PageViewController(VirtualView, this.MauiContext);
-			ViewController = _pageViewController;
+			if (ViewController == null)
+				ViewController =  new PageViewController(VirtualView, this.MauiContext);
```

Comment:
Should we check? Maybe we should just overwrite?
This is our creation, so why should we use what was set? What happens if I set some RandoViewController.? Do we expect some logic to exist in the PageViewController? We got theme and measure/arrange logic?

What happens if I stick say a generic VC in there? The like below will throw, so maybe we should always just override? If people want a new VC, they must override this method?

Reply:
> What happens if I stick say a generic VC in there?
I updated it so you could use a VC that doesn't inherit from PVC but it still wants `ContentView`. 

The `PageHandler` is typed to a `ContentView` and inherits from `ContentViewHandler`. It feels like we'd have to change a lot and it'd be tricky to just let this be setup so users can return an arbitrary View. 

But maybe @hartez has a better thought here.

> If people want a new VC, they must override this method?

Things like hot reload need to be able to weave a ViewController into the handlers.  Right now we have  `PageViewController : ContainerViewController` so the idea here with this would be that hotreload could weave into the ViewController through CreateNativeView and supply a new VC.  

> The like below will throw, so maybe we should always just override? If people want a new VC, they must override this method?

When we talked about the way we wanted factories to work the choice was that users would wire into `CreateNativeView` or the associated Mapper (i.e. `ItemSource`) and set public properties.